_studio/mfx_lib/shared/src/mfx_h264_encode_vaapi.cpp

```diff
@@ -1534,25 +1524,26 @@ mfxStatus VAAPIEncoder::CreateAuxilliaryDevice(
     m_caps.ddi_caps.ChromaWeightedPred      = 1;
     m_caps.ddi_caps.MaxNum_WeightedPredL0   = 4;
     m_caps.ddi_caps.MaxNum_WeightedPredL1   = 2;
-    m_caps.ddi_caps.Color420Only            = 1;
 
-    if ((attrs[idx_map[VAConfigAttribMaxPictureWidth]].value != VA_ATTRIB_NOT_SUPPORTED) &&
-        (attrs[idx_map[VAConfigAttribMaxPictureWidth]].value != 0))
-        m_caps.ddi_caps.MaxPicWidth  = attrs[idx_map[VAConfigAttribMaxPictureWidth]].value;
+    m_caps.ddi_caps.Color420Only = !(AV(VAConfigAttribRTFormat) & (VA_RT_FORMAT_YUV422 | VA_RT_FORMAT_YUV444));
+
+    if ((AV(VAConfigAttribMaxPictureWidth) != VA_ATTRIB_NOT_SUPPORTED) &&
```

Comment:
two or three lambda calls for proper value assignment... maybe it worth to rewrite?
```
m_caps.ddi_caps.MaxPicWidth = AV(VAConfigAttribMaxPictureWidth);

if (m_caps.ddi_caps.MaxPicWidth == VA_ATTRIB_NOT_SUPPORTED || m_caps.ddi_caps.MaxPicWidth == 0)
    m_caps.ddi_caps.MaxPicWidth = 1920;
```

Reply:
ok